language/move-prover/bytecode/src/access_path_trie.rs

```diff
@@ -0,0 +1,342 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    access_path::{AbsAddr, AccessPath, AccessPathMap, Offset, Root},
+    dataflow_analysis::{AbstractDomain, JoinResult, MapDomain},
+    stackless_bytecode::TempIndex,
+};
+use move_model::{model::FunctionEnv, ty::Type};
+use std::{
+    collections::btree_map::Entry,
+    fmt,
+    fmt::Formatter,
+    ops::{Deref, DerefMut},
+};
+
+/// The obvious approach to abstracting a set of concrete paths is using a set of abstract paths.
```

Comment:
Interesting question whether using a trie here instead of a flat set with redundancy is worth the complexity. I would expect 95% of the paths to be of length 2-4 (?). Or does the trie also helpful for the algorithms?